### PR TITLE
Bugfix/Feature: broken windows: dev build and publish

### PR DESCRIPTION
**Pull request by jackblackCH**
**Lerna / Monorepo architecture and build**

This PR improves the broken windows within our builds and components.

A fundamental part is the strict usage of fully independent components, which is needed for a proper use of a monorepo managed by Lerna, a tool that's really good at handling and connecting independent packages (i.e axa components).

Since we follow the rules, we can now:
- build and watch independent components/packages
- build and watch all components in parallel (Performance!)
- test (un)finished components without publishing them to NPM / aka "debugging in production"

**TLDR:**
The repository and architecture are now Lerna conform. Many components are refactored back into black boxes (no outer scope), so Lerna can do it's best job.
Generally, we have to imagine, that components live in their own repository (There is no outer scope). This forces that every component must have its own build and other tasks, configs can obviously be shared though!. 

If we follow these rules: Lerna can do the annoying parts of connecting the "loose" components. I.e, running builds in parallel, watch tasks in parallel, publish all or certain, versioning, and much more.
https://github.com/lerna/lerna

**Changes:**
- All components are "Lerna" conform, meaning:
   - have their own build
   - have their own watch
- Icon Build has moved to @axa-materials (no global script anymore)
  - Icon Build can be run independently as a component
- Improves @axa/materials
  - Export of raw sass styles
- Cleanup of global npm scripts ( less scripts, since many of them live in components)
- Features to test components ( see commands below)

**We can run now:** (more in the readmes)
- `npm run build` --> builds all components
- `npm run watch` --> watches all components in parallel (creates lib and dist build on files changes)
- test/debug components from the pattern lib in "production" across repositories via (npm link), more here: CONTRIBUTION.md
- Also improved -> https://github.com/axa-ch/patterns-library-examples
   - View styling: Screenshot 2019-05-27 at 20 55 21
   - Docs: How to test new components: 

**Improves global build time more than 50%, from ~90s to ~40s with 4 cores (thanks to Lerna --parallel flag)**

Screenshot 2019-05-27 at 21 06 18

### CODE REVIEW

**Comment by LucaMele** [comment]
I suggest we merge footer, top-content-bat and input-file, then rebase this, add the new jobs to package.json and scaffolding and all of us help test this PR.

File: .prettierignore [hunk 1]
```diff
@@ -1,7 +1,3 @@
-# Ignore everything except the src folder:
-/*
-!/src
```
Comment (LucaMele) [inline .prettierignore]: we said we do ONLY prettify src agreed in the team. Please revert

File: src/components/00-materials/README.md [hunk 2]
```diff
@@ -2,24 +2,47 @@
...
 ### Usage
 
-`npm install @axa-ch/materials`
+## Install Icons
 
-## Icon set
+`npm install @axa-ch/materials`
 
-### Usage
+## Useage
```
Comment (LucaMele) [inline src/components/00-materials/README.md]: Typo: Usage: https://dictionary.cambridge.org/dictionary/english/usage

File: src/components/00-materials/package.json [hunk 3]
```diff
@@ -20,10 +20,10 @@
     "type": "git",
     "url": "git+https://github.com/axa-ch/patterns-library.git"
   },
-  "scripts": {
-    "test": "echo \"Error: run tests from root\" && exit 1"
-  },
   "bugs": {
     "url": "https://github.com/axa-ch/patterns-library/issues"
+  },
+  "dependencies": {
```
Comment (LucaMele) [inline src/components/00-materials/package.json]: defently not a dependency but a devDependency

File: .prettierignore (hunk 1 above)
Comment (jackblackCH) [reply .prettierignore]: done

File: src/components/00-materials/package.json (hunk 3 above)
Comment (jackblackCH) [reply src/components/00-materials/package.json]: Nice catch. done.

File: src/components/00-materials/README.md (hunk 2 above)
Comment (jackblackCH) [reply src/components/00-materials/README.md]: Done.

File: src/components/00-materials/package.json (hunk 3 above)
Comment (jackblackCH) [reply src/components/00-materials/package.json]: done.

File: src/components/00-materials/README.md (hunk 2 above)
Comment (jackblackCH) [reply src/components/00-materials/README.md]: done.

File: .prettierignore (hunk 1 above)
Comment (jackblackCH) [reply .prettierignore]: done.

**Pull request merged**